cpp/src/copying/contiguous_split.cu

```diff
@@ -0,0 +1,550 @@
+static constexpr size_t split_align = 8;
+
+template<typename T>
+struct column_buf_size_functor_impl {
+   void operator()(column_view const& c, size_t& running_data_size, size_t& running_validity_size)
```

Comment:
Don't use pass by reference for outputs. Outputs should be _returned_.

Reply:
But in this case, it's only filling in part of a vector.  Doesn't it make more sense to have a single vector that gets filled in with multiple function calls rather than allocating mutiple vectors as return values and stitching them together?

Reply:
See my comment about using `std::accumulate` in `allocate_and_copy`. This function would be made to return a `std::pair<std::size_t, std::size_t>`.